cdc/log.cc

```diff
@@ -1258,6 +1258,52 @@ class transformer final : public change_processor {
     // Takes and returns generated cdc log mutations and associated statistics about parts touched during transformer's lifetime.
     // The `transformer` object on which this method was called on should not be used anymore.
     std::tuple<std::vector<mutation>, stats::part_type_set> finish() && {
+        static const auto preimg_op_bytes  = _op_col.type->decompose(operation_native_type(operation::pre_image));
+        static const auto postimg_op_bytes = _op_col.type->decompose(operation_native_type(operation::post_image));
+
+        // Adjust or remove deltas according to `delta_mode` in `cdc_options`.
+        if (_schema->cdc_options().get_delta_mode() != cdc::delta_mode::full) {
```

Comment:
I think it would be better if you place the adjust_or_remove in a separate function.
This way you could get rid of some of this right-drift using early returns instead of blocks under `if`s.

Reply:
done